ttax/decompositions.py

```diff
@@ -0,0 +1,181 @@
+import numpy as np
+import jax
+import jax.numpy as jnp
+
+from ttax.base_class import TT
+from ttax.base_class import TTMatrix
+
+def tt_round(tt, max_tt_rank=None, epsilon=None, name='ttax_round'):
+  """TT-rounding procedure, returns a TT object with smaller TT-ranks.
+  Args:
+    tt: `TensorTrain` object, TT-tensor or TT-matrix
+    max_tt_rank: a number or a list of numbers
+      If a number, than defines the maximal TT-rank of the result.
+      If a list of numbers, than `max_tt_rank` length should be d+1
+      (where d is the rank of `tens`) and `max_tt_rank[i]` defines
+      the maximal (i+1)-th TT-rank of the result.
+      The following two versions are equivalent
+        `max_tt_rank = r`
+      and
+        `max_tt_rank = r * np.ones(d-1)`
+    epsilon: a floating point number or None
+      If the TT-ranks are not restricted (`max_tt_rank=np.inf`), then
+      the result would be guarantied to be `epsilon` close to `tt`
+      in terms of relative Frobenius error:
+        ||res - tt||_F / ||tt||_F <= epsilon
+      If the TT-ranks are restricted, providing a loose `epsilon` may
+      reduce the TT-ranks of the result.
+      E.g.
+        round(tt, max_tt_rank=100, epsilon=0.9)
+      will probably return you a TT-tensor with TT-ranks close to 1, not 100.
+      Note that providing a nontrivial (= not equal to None) `epsilon` will make
```

Comment:
Change this to


Note that providing a nontrivial (= not equal to None) `epsilon` will make
      the TT-ranks of the result change depending on the data, which will prevent you from using jax.jit for speeding up the computations